### PR TITLE
add proxy component to ksonnet registry

### PR DESCRIPTION
**Pull request by IronPan**
This change add the proxy agent to the kubeflow cluster as part of gcp suit.
This change doesn't include the change to the new deployment go binary. I'll add it in a separate PR

Currently waiting for image to be released

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/kubeflow/2746)

### CODE REVIEW

File: bootstrap/config/kfctl_basic_auth.yaml [hunk 1]
```diff
@@ -28,7 +28,7 @@ components:
   - basic-auth
   - basic-auth-ingress
   - centraldashboard
-  - cert-manager
+  - proxy-agent
```
Comment (kunmingg) [inline bootstrap/config/kfctl_basic_auth.yaml]: You can add another entry to config folder like 
kfctl_proxy.yaml

File: components/gcp-click-to-deploy/app-config.yaml [hunk 2]
```diff
@@ -18,6 +18,8 @@ defaultApp:
     prototype: cert-manager
   - name: iap-ingress
     prototype: iap-ingress
+  - name: proxy-agent
+    prototype: proxy-agent
```
Comment (kunmingg) [inline components/gcp-click-to-deploy/app-config.yaml]: proxy-agent can be an optional components based on user's choice of how to access UI.
We should make the change on js side.

Comment (IronPan) [reply components/gcp-click-to-deploy/app-config.yaml]: is this file for local test only? if not adding it, will local test be able to deploy proxy agent component?

File: bootstrap/config/kfctl_basic_auth.yaml (hunk 1 above)
Comment (IronPan) [reply bootstrap/config/kfctl_basic_auth.yaml]: done

File: bootstrap/config/kfctl_proxy.yaml [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+  - seldon
+  - tensorboard
+  - tf-serving
+  - tf-training
+components:
+  - ambassador
+  - argo
+  - centraldashboard
+  - cert-manager
+  - cloud-endpoints
```
Comment (kunmingg) [inline bootstrap/config/kfctl_proxy.yaml]: We might just remove cert-manager and cloud-endpoints as well.

File: bootstrap/config/kfctl_proxy.yaml [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+  - notebook-controller
+  - pipeline
+  - profiles
+  - proxy-agent
+  - pytorch-operator
+  - spartakus
+  - tensorboard
+  - tf-job-operator
+componentParams:
+  cert-manager:
```
Comment (kunmingg) [inline bootstrap/config/kfctl_proxy.yaml]: Same here and below

File: components/gcp-click-to-deploy/app-config.yaml (hunk 2 above)
Comment (kunmingg) [reply components/gcp-click-to-deploy/app-config.yaml]: Yeah, this one is for local test only.
We can edit config in javascript based on a param?

File: scripts/gke/util.sh [hunk 5]
```diff
@@ -246,7 +247,7 @@ gcpKsApply() {
   createKsEnv
 
   if [[ -z $DEFAULT_KUBEFLOW_COMPONENTS ]]; then
-    export KUBEFLOW_COMPONENTS+=',"cloud-endpoints","cert-manager","iap-ingress"'
+    export KUBEFLOW_COMPONENTS+=',"cloud-endpoints","cert-manager","iap-ingress", "proxy-agent"'
```
Comment (kunmingg) [inline scripts/gke/util.sh]: If use ```proxy-agent```, then ```cloud-endpoints","cert-manager","iap-ingress``` is not needed.
Can we control by parameter?
Or we can just implement the logic in kfctl-go which should be easier

**Comment by jlewi** [comment]
@IronPan Can we close this PR?

**Pull request closed without merging**